plugins/ipc/lab.py

```diff
@@ -0,0 +1,174 @@
+"""
+Helpers for working with lab tests for the purposes of IPC
+"""
+from plugins.labtests import models as lab_test_models
+
+from plugins.ipc import models
+
+
+
+class IPCTest(object):
+
+    TEST_NAME         = None # Name of the test (human readable & used to filter)
+    TEST_CODE         = None # Winpath test code used for e.g. urls
+    OBSERVATION_NAMES = []   # Names of the observation used to classify this test
+    ALERT_CATEGORY    = None # The category of alert to use when creating an InfectionAlert for this test
+
+    def __init__(self, test):
+        self.test    = test
+        self.patient = test.patient
+
+    def __repr__(self):
+        return "\n".join(
+            [f"{self.test.lab_number} {self.test.test_code} {self.test.test_name} {self.test.datetime_ordered}"] +
+            [f"{o.observation_name} {o.observation_value}" for o in self.get_observations()]
+        )
+
+    def get_observations(self):
+        return self.test.observation_set.filter(
+            observation_name__in=self.OBSERVATION_NAMES
+        ).exclude(observation_value='.{}')
+
+    def alert(self):
+        """
+        Is this test an alert result?
+        """
+        raise NotImplementedError()
+
+
+class CarbapenemaseScreen(IPCTest):
+
+    TEST_NAME         = 'CARBAPENEMASE SCREEN'
+    TEST_CODE         = 'CARS'
+    OBSERVATION_NAMES = ['Carbapenemase screen culture']
+    ALERT_CATEGORY    = models.InfectionAlert.CPE
+
+    def alert(self):
+        """
+        If the Screen grew something, return True otherwise return False
+        """
+        for observation in self.get_observations():
+            if observation.observation_value.startswith('1'):
+                return True
+        return False
+
+
+class CDiffPCR(IPCTest):
+
+    TEST_NAME         = 'CDIFF PCR AND TOXIN'
+    TEST_CODE         = 'DIFF'
+    OBSERVATION_NAMES = ['C.diff PCR test', 'C.diff Toxin test']
+    ALERT_CATEGORY    = models.InfectionAlert.CDIFF
+
+    def alert(self):
+        """
+        If either observation is Positive, return True
+        """
+        for observation in self.get_observations():
+            if observation.observation_value == 'POSITIVE':
```

Comment:
based on my local DB are you sure we don't want observation.observation_value.startswith('POSITIVE')?

Reply:
Possibly, but we're reviewing this with clinical team later this week - right now these are indicative only